test/framework/src/main/java/org/elasticsearch/test/ESAllocationTestCase.java

```diff
@@ -196,6 +196,32 @@ public Decision canAllocate(RoutingNode node, RoutingAllocation allocation) {
         }
     }
 
+    /**
+     * A decider class that allows controlling the decision on both allocating (AllocationDecider#canAllocate)
+     * and force allocating (AllocationDecider#canForceAllocatePrimary) a primary shard.
+     */
+    protected static class ForcePrimaryDecider extends TestAllocateDecision {
```

Comment:
Instead of having this class, it might be more readable to just have two methods in `PrimaryShardAllocatorTests`:

```
AllocationDecider getNoDeciderThatThrottlesForceAllocate() {
   return new TestAllocateDecision(Decision.NO) {
       @Override
       public Decision canForceAllocatePrimary(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
            return Decision.THROTTLE;
       }
   };
}

AllocationDecider getNoDeciderThatAllowsForceAllocate() {
     ... return YES;

}
```